frontend/features/plugins/usePagePlugin.ts

```diff
@@ -72,50 +71,69 @@ function getPageInput(data: PageData): UpdatePageInput {
     data: {
       pageName: data.title,
       path: data.path,
-      blocks: data.blocks.map((block) => {
-        console.log("Blocks", data.blocks);
-
-        switch (block._template) {
-          case "hero": {
+      sections: data.sections.map((section) => {
+        switch (section._template) {
+          case "heroSection": {
             return {
-              __typename: "ComponentBlocksHero",
-              id: block.id,
-              title: block.title,
-              subtitle: block.subtitle,
+              __typename: "ComponentSectionHeroSection",
+              id: section.id,
+              title: section.title,
+              subtitle: section.subtitle,
             };
           }
-          case "card": {
+          case "cardSection": {
             return {
-              __typename: "ComponentBlocksCard",
-              id: block.id,
-              title: block.title,
-              description: block.description,
-              projectLink: block.projectLink,
-              imageUrl: block.imageUrl ? block.imageUrl : "undefined",
+              __typename: "ComponentSectionCardSection",
+              id: section.id,
+              title: section.title,
+              subtitle: section.subtitle,
+              sectionTitle: section.sectionTitle,
+              sections: section.blocks?.map((card) => {
+                if (card != null) {
+                  return {
+                    id: card.id,
+                    title: card.title,
+                    description: card.description,
+                    imageUrl: card.imageUrl,
+                    projectLink: card.projectLink,
+                  };
+                }
```

Comment:
Here we should filter out null values from the blocks array before doing the map